src/components/memorygame/MemoryGameCards/index.jsx

```diff
@@ -0,0 +1,171 @@
+import memoryGameSuccessSymbol from "../../../images/memoryGameSuccessSymbol.svg";
+import cardInitialStateGenerator from "../cardInitialStateGenerator";
+
+const NUM_REQUIRED_MATCHES = 2;
+// game example: https://www.brightfocus.org/alzheimers/memory-games/match-pictures
+// instead of squares this game uses cards.
+// onClick card flips to show what is behind the card.
+// matchedCards is an array states that holds successfully matched card states
+// setCompleted is boolean
+// setCompleted is true when all possible matches made
+const MemoryGameCards = ({ setMatchedCards, matchedCards, setCompleted }) => {
+  const [t] = useTranslation();
+  // cardsDataArr example [{description:"...",link:"...",phrase:".."},...]
+  const cardsDataArr = [
+    ...t("memoryGame.cards", {
+      returnObjects: true,
+    }),
+  ];
+  // cardStates is an array of states with multiple properties
+  // state example :
+  // {cardKey:"",img:{src:"",imgKey:""},
+  // isFlipped:"",isMatched:"",description:"",link:"",phrase:""}
+  const [cardStates, setCardStates] = useState(
+    cardInitialStateGenerator(cardsDataArr, NUM_REQUIRED_MATCHES)
+  );
+  // selectedCards is an array of clicked card states(objects).
+  // selectedCards array hold only "NUM_REQUIRED_MATCHES" amount of card states
+  const [selectedCards, setSelectedCards] = useState([]);
+
+  // handle click flips card and adds that card's state to selectedCards
+  const handleClick = (clickedCardState) => {
+    const copyOfCardStates = [...cardStates];
+    // current index is needed to keep the order of cards displayed
+    // because goal is to make user remember the spots of previous cards
+    const currentIndex = cardStates.findIndex(
+      (state) => state.cardKey === clickedCardState.cardKey
+    );
+    copyOfCardStates[currentIndex].isFlipped = true;
+    setCardStates(() => [...copyOfCardStates]);
+    setSelectedCards((prevState) => [...prevState, clickedCardState]);
+  };
+
+  // matched cards should have different cardKey, same image key, isFlipped:true
+  // function below tests it
+  const doesMatchLastCard = (prevCardState, lastCardState) => {
+    return (
+      prevCardState.cardKey !== lastCardState.cardKey &&
+      prevCardState.img.imgKey === lastCardState.img.imgKey &&
+      prevCardState.isFlipped &&
+      lastCardState.isFlipped
+    );
+  };
+
+  // Triggered when user clicks on a card(selectedCards change)
+  // checks selected cards to see if they are a match
+  useEffect(() => {
+    // only triggered if "NUM_REQUIRED_MATCHES" amount of cards are clicked
+    if (selectedCards.length !== NUM_REQUIRED_MATCHES) {
+      return;
+    }
+    // match condition of selectedCards will be based on card that is added last to selectedCards
+    const lastCardState = selectedCards[selectedCards.length - 1];
+    const matchesOfLastCardState = selectedCards.filter((prevCard) =>
+      doesMatchLastCard(prevCard, lastCardState)
+    );
```

Comment:
rename doesMatchLastCard to:
`areCardsMatching`

with a comment
```
// Checks if two cards have a matching image.
```

you should also be able to delete line 55/56 the flipped conditions. this function doesn't care if its flipped; all selectedCards should be flipped by the time its in line 63 as far as i know. its not possible to be a selected card and not flipped, right? (duplicate checks like these lead to bugs)

```
// Check all cards correspond to the same image.
selectedCards.every((c) => {
   return areCardsMatching(c, selectedCards[0])
})
```

then you can clean up this code a bunch and delete lastCardState, matchesOfLastCardState, matchedCardsArr

Reply:
oh right. Before I had automatic unflip mechanism. That's why I still have flip condition

Reply:
`.every` is a useful method. I implemented and adjusted the code accordingly